clients/fides-js/src/fides.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * Fides.js: Javascript library for Fides (https://github.com/ethyca/fides)
+ *
+ * This JS module provides easy access to interact with Fides from a webpage, including the ability to:
+ * - initialize the page with default consent options (e.g. opt-out of advertising cookies, opt-in to analytics, etc.)
+ * - read/write the current user's consent preferences to their browser as a cookie
+ * - push the current user's consent preferences to other systems via integrations (Google Tag Manager, Meta, etc.)
+ *
+ * See https://fid.es for more information!
+ *
+ * Basic usage of this module in an HTML page is:
+ * ```
+ * <script src="https://privacy.{company}.com/fides.js"></script>
+ * <script>
+ *   window.Fides.init({
+ *     consent: {
+ *       options: [{
+ *         cookieKeys: ["data_sales"],
+ *         default: true,
+ *         fidesDataUseKey: "advertising"
+ *       }]
+ *     }
+ *   });
+ * </script>
+ * ```
+ *
+ * ...and later:
+ * ```
+ * <script>
+ *   // Query user consent preferences
+ *   if (window.Fides.consent.data_sales) {
+ *     // ...enable advertising scripts
+ *   }
+ * ```
+ */
+import { gtm } from "./integrations/gtm";
+import { meta } from "./integrations/meta";
+import { shopify } from "./integrations/shopify";
+import { ConsentConfig } from "./lib/consent-config";
+import { getConsentContext } from "./lib/consent-context";
+import { initFidesConsent } from "./lib/consent";
+import {
+  CookieKeyConsent,
+  CookieIdentity,
+  CookieMeta,
+  getOrMakeFidesCookie,
+  makeConsentDefaults,
+} from "./lib/cookie";
+import { ConsentBannerOptions } from "./lib/consent-types";
+import { getBannerOptions } from "./lib/consent-utils";
+
+export interface FidesConfig {
+  consent: ConsentConfig;
+  bannerOptions?: ConsentBannerOptions;
+}
```

Comment:
New strong opinion here!

I'd suggest we be super clear and opinionated on this and re-think the config options for this package into:
```
export interface FidesConfig {
  consent: ConsentConfig; // (optional) set the consent defaults from a "legacy" Privacy Center config.json
  experience: ExperienceConfig; // (optional) set the "experience" to be used for this Fides.js instance -- overrides the "legacy" config
  geolocation: UserGeolocation; // (optional) set the geolocation for this Fides.js instance
  options: FidesOptions; // (optional) global options for this Fides.js instance
}
```

I think this'll make it much easier to test, configure, and migrate to if we tinker it this way...